pkg/http/http.go

```diff
@@ -63,6 +80,106 @@ func (b BasicAuth) IsZero() bool {
 	return b.Username == "" && b.Password == "" && b.PasswordFile == ""
 }
 
+func NewClientConfigFromYAML(cfg []byte) (*ClientConfig, error) {
+	conf := &ClientConfig{}
+	if err := yaml.UnmarshalStrict(cfg, conf); err != nil {
+		return nil, err
+	}
+	return conf, nil
+}
+
+type httpClientOptions struct {
+	maxIdleConns        int
+	maxIdleConnsPerHost int
+}
+
+type HTTPClientOption func(options *httpClientOptions)
+
+func WithMaxIdleConns(maxIdleConns int) HTTPClientOption {
+	return func(opts *httpClientOptions) {
+		opts.maxIdleConns = maxIdleConns
+	}
+}
+
+func WithMaxIdleConnsPerHost(maxIdleConnsPerHost int) HTTPClientOption {
+	return func(opts *httpClientOptions) {
+		opts.maxIdleConnsPerHost = maxIdleConnsPerHost
+	}
+}
+
+// NewRoundTripperFromConfig returns a new HTTP RoundTripper configured for the
+// given http.HTTPClientConfig and http.HTTPClientOption.
+func NewRoundTripperFromConfig(cfg config_util.HTTPClientConfig, name string, optFuncs ...HTTPClientOption) (http.RoundTripper, error) {
+	opts := defaultHTTPClientOptions
+	for _, f := range optFuncs {
+		f(&opts)
+	}
+
+	newRT := func(tlsConfig *tls.Config) (http.RoundTripper, error) {
+		var rt http.RoundTripper = &http.Transport{
+			Proxy:               http.ProxyURL(cfg.ProxyURL.URL),
+			MaxIdleConns:        opts.maxIdleConns,
+			MaxIdleConnsPerHost: opts.maxIdleConnsPerHost,
+			DisableKeepAlives:   true,
+			TLSClientConfig:     tlsConfig,
+			DisableCompression:  true,
```

Comment:
I think we default to this because there was 'something with it' for Thanos side (_citation needed_). As this is more a generic package we should default to `false` and explicitly set this too `true` on functionalities via arguments that require this to be disabled. 

Tl;dr IMO:  set 'defaults' to something that would be defaults for _all_ packages, and override different cases when needed. Not the other way around. :)

Reply:
We can, but we need to be aware who is using `exthttp.NewTransport` and how